### PR TITLE
To-Do List - Code Review

### PR DESCRIPTION
**Pull request by emiliocm9**
# To-Do List in JavaScript
In this PR I have implemented the features to comply with the project requirements. The following list includes some of them:

- [x] Users can create groups with a name and a description.
- [x] Users can create tasks and attach them to a certain group with a title, description, date, completion, and level of priority.
- [x] Users can edit or delete any of the tasks.
- [x] LocalStorage is used to keep the information in the browser.
- [x] Bootstrap is used to improve the design.
- [x] This project has been launched using Github Pages. [link](https://emiliocm9.github.io/To-Do-List-JavaScript/)

>This PR is going to be reviewed by TSE's from Microverse.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    $('.menumobile').toggleClass('menuoff');
+    $('.menutoggle').toggleClass('buttonmove');
+  });
+});
+
+uploadProject();
+initFun();
+hideEditTask();
+
+window.onload = () => {
```
Comment (mosaaleb) [inline src/index.js:20]: - [x] I got a console error here:

Screenshot from 2020-11-12 11-52-57

Let's fix this.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* eslint-env jquery */
+import { initFun } from './projectmodules';
+import uploadProject from './projectcreate';
+import { hideEditTask } from './projectselect';
+
+$(() => {
```
Comment (mosaaleb) [inline src/index.js]: - [x] Using JQuery isn't allowed in JS projects. Let's convert this to pure Javascript.

File: src/cardsdisplay.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    myContainer.appendChild(card);
+  } else {
+    checkList.checked = false;
+    undoneContainer.appendChild(card);
+  }
+};
+
+const deleteTask = (project, myCard, cardIndex) => {
+  myCard.querySelector('#delete').addEventListener('click', () => {
+    if (window.confirm('Are you sure you want to delete this task?')) {
```
Comment (mosaaleb) [inline src/cardsdisplay.js]: - [x] Do not use Window.alert() or Window.confirm(). It's impossible to style, stop code execution and display differently on different browsers. Use custom modal instead.

File: src/cardsdisplay.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  });
+};
+
+const editTask = (project, myCard, cardIndex) => {
+  myCard.querySelector('#edit').addEventListener('click', () => {
+    selectContainer.classList.add('d-none');
+    selectContainerLabel.classList.add('d-none');
+    taskbutton.classList.remove('d-none');
+    savebutton.classList.add('d-none');
+    $('#exampleModal').modal('show');
```
Comment (mosaaleb) [inline src/cardsdisplay.js]: - [x] Lte's replace this with pure JS.

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Changes required


Amazing work @emiliocm9 :clap:. You have done nearly all the requirements, yet there are still some to work on before moving on:


- [x] There should be a default project which combines all todos from all projects



- [ ] **[OPTIONAL]** It would be nice if we have added a feature to delete/edit a project.


Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @emiliocm9 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**